ideascube/serveradmin/views.py

```diff
@@ -111,3 +112,47 @@ def backup(request):
 def battery(request):
     return render(request, 'serveradmin/battery.html',
                   {'batteries': batinfo.batteries()})
+
+
+@staff_member_required
+def wifi(request, ssid=''):
+    wifi_list = AvailableWifiNetwork.all()
+
+    if ssid:
+        for wifi in wifi_list:
```

Comment:
Seems that you are often looping through the wifi list to match the ssid: I think that would at least deserve a method of the AvailableWifiNetwork (something like `get(ssid)`), plus it may be worth an OrderedDict instead of a list, so we don't have to loop.